src/client/opamSwitchCommand.ml

```diff
@@ -392,6 +392,13 @@ let switch lock gt switch =
     (OpamSwitch.to_string switch) (OpamSwitch.to_string switch)
     (OpamStd.Format.itemize OpamSwitch.to_string installed_switches)
 
+let switch_previous lock gt =
+  match OpamFile.Config.previous_switch gt.config with
+  | Some switch_name -> switch lock gt switch_name
```

Comment:
should we check here previous switch existence, in order to display a more specific error message than `switch` one?

Reply:
I don’t think it’s necessary, the default error message from opam switch is perfectly fine i think

Reply:
```suggestion
  | Some switch_name ->
    if not (OpamGlobalConfig.switch_exists gt switch_name) then
      OpamConsole.error_and_exit `Not_found "The previous switch no longer exists";
    switch lock gt switch_name
```
Something like that?